test/input/yaml/CustomInput_routelink_csv_alternative.yaml

```diff
@@ -0,0 +1,82 @@
+      nc_output_folder: "../../test/output/text"
+  #data column assignment inside supernetwork_parameters
+  supernetwork_parameters:
+      title_string: Custom Input Example
+      geo_file_path: "../../test/input/geo/wrf-hydro-run/DOMAIN/RouteLink.nc"
+      #mask_file_path: "RnR_small_domain.csv"
+      #mask_layer_string: ""
+      #mask_driver_string: "csv"
+      #mask_key: 0
+      columns:
+          key: "link"
+          downstream: "to"
+          dx: "Length"
+          n: "n"  # TODO: rename to `manningn`
+          ncc: "nCC"  # TODO: rename to `mannningncc`
+          s0: "So"  # TODO: rename to `bedslope`
+          bw: "BtmWdth"  # TODO: rename to `bottomwidth`
+          waterbody: "NHDWaterbodyComID"
+          tw: "TopWdth"  # TODO: rename to `topwidth`
+          twcc: "TopWdthCC"  # TODO: rename to `topwidthcc`
+          musk: "MusK"
+          musx: "MusX"
+          cs: "ChSlp"  # TODO: rename to `sideslope`
+      waterbody_null_code: -9999
+      terminal_code: 0
+      driver_string: NetCDF
+      layer_string: 0
+
+  # #WRF-Hydro output file
+  forcing_parameters:
+      qlat_input_folder: "../../test/input/geo/wrf-hydro-run/OUTPUTS/"
+      qlat_file_pattern_filter: "/*.CHRTOUT_DOMAIN1"
+      qlat_file_index_col: station_id
+      qlat_file_value_col: q_lateral
+  #WRF-Hydro restart files
+  restart_parameters:
+      #WRF-Hydro channels restart file
+      wrf_hydro_channel_restart_file: "../../test/input/geo/wrf-hydro-run/restart/HYDRO_RST.2020-03-19_18:00_DOMAIN1"
+      #WRF-Hydro channels ID crosswalk file
+      wrf_hydro_channel_ID_crosswalk_file: "../../test/input/geo/wrf-hydro-run/DOMAIN/RouteLink.nc"
+      wrf_hydro_channel_ID_crosswalk_file_field_name: link
+      wrf_hydro_channel_restart_upstream_flow_field_name: qlink1
+      wrf_hydro_channel_restart_downstream_flow_field_name: qlink2
+      wrf_hydro_channel_restart_depth_flow_field_name: hlink
+      # #WRF-Hydro waterbodies restart file
+      # wrf_hydro_waterbody_restart_file: "../../test/input/geo/NWM_2.1_Sample_Datasets/Pocono_TEST1/example_RESTART/HYDRO_RST.2017-12-31_06-00_DOMAIN1"
+      # #WRF-Hydro waterbody ID crosswalk file
+      # wrf_hydro_waterbody_ID_crosswalk_file: "../../test/input/geo/NWM_2.1_Sample_Datasets/Pocono_TEST1/primary_domain/DOMAIN/LAKEPARM.nc"
+      # wrf_hydro_waterbody_ID_crosswalk_file_field_name: lake_id
+      # #WRF-Hydro waterbody crosswalk filter file
+      # wrf_hydro_waterbody_crosswalk_filter_file: "../../test/input/geo/NWM_2.1_Sample_Datasets/Pocono_TEST1/primary_domain/DOMAIN/Route_Link.nc"
+      # wrf_hydro_waterbody_crosswalk_filter_file_field_name: NHDWaterbodyComID
+  parity_parameters:
+      parity_check_input_folder: "../../test/input/geo/wrf-hydro-run/OUTPUTS/"
+      parity_check_file_pattern_filter: "/*.CHRTOUT_DOMAIN1"
+      parity_check_file_index_col: station_id
+      parity_check_file_value_col: streamflow
+      parity_check_compare_node: 20331516
+  data_assimilation_parameters:
```

Comment:
This should be an either/or

Either we use csv (defined by one file) or we use the wrf-hydro-style timeslices in a folder (defined by three parameters)